llama_stack/apis/common/content_types.py

```diff
@@ -107,10 +109,34 @@ class ToolCallDelta(BaseModel):
     parse_status: ToolCallParseStatus
 
 
+@json_schema_type
+class InProgressToolCallDelta(BaseModel):
```

Comment:
since we have InProgressToolCallDelta, should we also add a `FailedToolCallDelta` for a place to set messages if a tool call parse fails?


E.g. https://github.com/meta-llama/llama-stack/blob/aebd130b082814a5433f04dd9a4f50d873068b5c/llama_stack/providers/utils/inference/openai_compat.py#L364-L374

Reply:
@yanxi0830 now we don't need that. `FinalToolCallDelta(tool_call=None)` is for that reason

Reply:
If we set tool_call=None, where do we put the raw message text from the failed tool call if the parse fails?

Reply:
we have already communicated that piece str-by-str earlier right?

Reply:
Yes, but that means we would have to re-loop and populate the buffer compared with just getting the message from the FinalToolCallDelta?

https://github.com/meta-llama/llama-stack/blob/aebd130b082814a5433f04dd9a4f50d873068b5c/llama_stack/providers/inline/agents/meta_reference/agent_instance.py#L513-L519